internal/github/github.go

```diff
@@ -392,23 +392,42 @@ func (c GHClient) GetCIStatus(ctx context.Context, owner, repoName, ref string)
 	}
 
 	if isFailure {
-		log.Debug("Failed CI checks:")
+		reportFailure := false
+		if len(excludes) == 0 {
+			reportFailure = true
+			log.Debug("Failed CI checks:")
+		}
+
 		for _, node := range nodes {
 			isCheckFailure := strings.ToLower(node.CheckRun.Conclusion) == "failure"
 			isStatusFailure := strings.ToLower(node.StatusContext.State) == "failure"
 
 			switch node.Typename {
 			case "CheckRun":
 				if isCheckFailure {
-					log.Debug(fmt.Sprintf("CheckRun '%s' failed", node.CheckRun.Name))
+					if !slices.Contains(excludes, node.CheckRun.Name) {
+						reportFailure = true
+						log.Debug(fmt.Sprintf("CheckRun '%s' failed", node.CheckRun.Name))
+					} else {
+						log.Debug(fmt.Sprintf("CheckRun '%s' failed but excluded", node.CheckRun.Name))
+					}
 				}
 			case "StatusContext":
 				if isStatusFailure {
-					log.Debug(fmt.Sprintf("StatusContext '%s' failed", node.StatusContext.Context))
+					if !slices.Contains(excludes, node.StatusContext.Context) {
+						reportFailure = true
+						log.Debug(fmt.Sprintf("StatusContext '%s' failed", node.StatusContext.Context))
+					} else {
+						log.Debug(fmt.Sprintf("StatusContext '%s' failed but excluded", node.StatusContext.Context))
+					}
 				}
 			}
 		}
-		return CIStatusFailed, nil
+		if reportFailure {
+			return CIStatusFailed, nil
+		} else {
+			return CIStatusPassed, nil
+		}
```

Comment:
Very nitpicky, but I have been told this is more inline with typical Go convention/idiomatic:

```suggestion
		if reportFailure {
			return CIStatusFailed, nil
		}
		return CIStatusPassed, nil
```